File: leetcode-algorithms/035. Search Insert Position/SearchInsertPosition.cs
public class Solution {
    public int SearchInsert(int[] nums, int target) {
        if(nums.Length == 0){
            return 0;
        }
        int i=0;
        for(;i<nums.Length;i++){
            if(target==nums[i]){
                return i;
            }
            if(target < nums[i]){
                return i;
            }
        }
        return i;
    }
}

File: leetcode-algorithms/035. Search Insert Position/SearchInsertPosition.py
class Solution:
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        #二分法 O(log n) time, O(1) space
        low = 0
        high = len(nums)
        if target <= nums[low]:
            return low
        if target > nums[high-1]:
            return high
        while low < high:
            mid = low + (high - low)//2
            if target < nums[mid]:
                high = mid
            elif target > nums[mid]:
                low = mid +1
            else:
                return mid
        return low